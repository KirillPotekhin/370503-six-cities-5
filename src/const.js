export const SortingOption = {
  POPULAR: `Popular`,
  PRICE_LOW_TO_HIGH: `Price: low to high`,
  PRICE_HIGH_TO_LOW: `Price: high to low`,
  TOP_RATED_FIRST: `Top rated first`,
};

export const SizePreviewImage = {
  WIDTH_PREVIEW_DEFAULT: `260`,
  HEIGHT_PREVIEW_DEFAULT: `220`,
  WIDTH_PREVIEW_FOR_FAVORITES: `150`,
  HEIGHT_PREVIEW_FOR_FAVORITES: `110`,
};

export const ReviewRatingTitle = {
  PERFECT: `perfect`,
  GOOD: `good`,
  NOT_BAD: `not bad`,
  BEADLY: `beadly`,
  TERRIBLY: `terribly`,
};

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};
