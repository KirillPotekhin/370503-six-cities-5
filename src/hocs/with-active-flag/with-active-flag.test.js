import React from "react";
import renderer from "react-test-renderer";
import PropTypes from "prop-types";
import withActiveFlag from "./with-active-flag";
import {AppRoute, AuthorizationStatus} from "../../const";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import {Route, Router as BrowserRouter, Switch} from "react-router-dom";
import browserHistory from "../../browser-history";

const MockComponent = (props) => {
  const {children} = props;

  return (
    <React.Fragment>
      {children}
    </React.Fragment>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};

const MockComponentWrapped = withActiveFlag(MockComponent);

describe(`test withActiveFlag`, () => {
  it(`test withActiveFlag!`, () => {
    const mockStore = configureStore([]);
    let store = null;
    let WithActiveFlagComponent = null;

    beforeEach(() => {
      store = mockStore({
        STATE: {
          city: {
            location: {
              latitude: 52.3909553943508,
              longitude: 4.85309666406198,
              zoom: 12,
            },
            name: `Amsterdam`,
          },
        },
        DATA: {
          offers: [],
          reviews: [],
          cities: [],
          offersNearby: [],
          offersFavorites: [],
        },
        USER: {
          authorizationStatus: AuthorizationStatus.NO_AUTH,
          email: ``,
        }
      });

      store.dispatch = jest.fn();

      WithActiveFlagComponent = renderer.create(
          <Provider store={store}>
            <BrowserRouter history={browserHistory}>
              <Switch>
                <Route exact path={AppRoute.LOGIN}>
                  <MockComponentWrapped />
                </Route>
              </Switch>
            </BrowserRouter>
          </Provider>
      );
    });

    expect(WithActiveFlagComponent).toMatchSnapshot();
  });
});
