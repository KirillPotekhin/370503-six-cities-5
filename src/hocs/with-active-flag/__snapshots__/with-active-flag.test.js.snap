// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test withActiveFlag test withActiveFlag! 1`] = `null`;
